editor/src/components/canvas/commands/utils/group-resize-utils.ts

```diff
@@ -0,0 +1,129 @@
+import type { LocalRectangle, Size } from '../../../../core/shared/math-utils'
+import { roundToNearestWhole } from '../../../../core/shared/math-utils'
+
+type FramePoint = keyof FrameWithAllPoints
+
+const AllFramePoints = ['left', 'top', 'right', 'bottom', 'width', 'height'] as const
+const HorizontalFramePoints = ['left', 'right', 'width'] as const
+const VerticalFramePoints = ['top', 'bottom', 'height'] as const
+
+function isHorizontalPoint(point: FramePoint): boolean {
+  return (HorizontalFramePoints as ReadonlyArray<FramePoint>).includes(point)
+}
+function isVerticalPoint(point: FramePoint): boolean {
+  return (VerticalFramePoints as ReadonlyArray<FramePoint>).includes(point)
+}
+
+export type FrameWithAllPoints = {
+  left: number
+  top: number
+  right: number
+  bottom: number
+  width: number
+  height: number
+}
+
+export function localRectangleToSixFramePoints(
+  localRectangle: LocalRectangle,
+  containerSize: Size,
+): FrameWithAllPoints {
+  return {
+    left: localRectangle.x,
+    right: containerSize.width - (localRectangle.x + localRectangle.width),
+    width: localRectangle.width,
+    top: localRectangle.y,
+    bottom: containerSize.height - (localRectangle.y + localRectangle.height),
+    height: localRectangle.height,
+  }
+}
+
+export function roundSixPointFrameToNearestWhole(
+  fullFrame: FrameWithAllPoints,
+): FrameWithAllPoints {
+  return {
+    left: roundToNearestWhole(fullFrame.left),
+    top: roundToNearestWhole(fullFrame.top),
+    right: roundToNearestWhole(fullFrame.right),
+    bottom: roundToNearestWhole(fullFrame.bottom),
+    width: roundToNearestWhole(fullFrame.width),
+    height: roundToNearestWhole(fullFrame.height),
+  }
+}
+
+export function transformConstrainedLocalFullFrameUsingBoundingBox(
+  newBoundingBox: Size,
+  currentBoundingBox: Size,
+  currentFrame: FrameWithAllPoints,
+  constrainedPoints: Array<FramePoint>,
+): FrameWithAllPoints {
+  // TODO verify that currentBoundingBox and currentFrame's dimensions match up
```

Comment:
note: I'm not exactly sure what this means - the `currentBoundingBox` relates to the group dimensions, while the `currentFrame` relates to the _child_ dimensions at the start of the interaction.
So I see alternative possible interpretations:
1. make sure that the computed sizes (at the caller side) _add up_ to the original bounding box
2. make sure that the final sizes (at the caller side) _add up_ to the final bounding box
3. possibly more?

Therefore I'm leaving this here until more light is cast on this